python/task_manager/background_task_manager.py

```diff
@@ -20,6 +20,11 @@
 from .worker_thread import WorkerThread
 from .results_poller import ResultsDispatcher
 
+# Set to True to enable extensive debug logging.
+# Useful for debugging concurrency issues.
+# Do not release with this setting set to True!
+ENABLE_DETAILED_DEBUG = False
```

Comment:
i have been to environment variable hell and back several times in my carreer. i think if the need ever arises, we can add it. One of Toolkit's original design goals was "no environment variables!" :)